### ISSUE TITLE
Identifying pairwise distances in `RadialModel`

### ISSUE THREAD

**Issue by bergdorf** [e3nn/e3nn#79]
In my usage I need to back out which points a pairwise distance is associated with, for instance so that I can scale it with a pair-wise parameter that I optimize. 
I noticed that with some of the recent changes, you started filtering out 
radii <= eps here
https://github.com/e3nn/e3nn/blob/master/e3nn/kernel.py#L122
This way the `RadialModel` is not guaranteed to receive all pairwise distances, 
and I can't trivially infer which i,j a distance corresponds to.

Two possible work-arounds come to mind:

1. One could add the `r_eps` parameter to `Convolution.forward` so it can be passed
   on to `Kernel.forward`, I'd then run it with `r_eps=-1`
2. One could pass the `radii > r_eps` mask to the `RadialModel`, as in 
   R = self.R(radii[radii > r_eps], mask=(radii > r_eps))

Do you have any advice?

P.s.: absolutely <3 this work of yours.

**Comment by mariogeiger** [comment]
I will add `r_eps` parameter to `Convolution.forward`.

I'm curious to know better what you are doing. I'm wondering if the network you create is still equivariant. It looks like it is not anymore permutation equivariant if the radial component depends on the pair of points.

**Comment by bergdorf** [comment]
However, do your RSH numerics require rij > 0 for numerical stability?

Indeed it is not permutation invariant anymore as my points are not identical: my points are atoms, each atom is of a certain type (element type for instance) and each atom-type-pair has its own scaling component. Thus the model is not globally permutation invariant, but still permutation invariant among all equal atom-type-pairs.

**Comment by mariogeiger** [comment]
Then you should have a different kernel per atom type pairs.

Here is an example of how I would manage that https://github.com/e3nn/e3nn/blob/master/examples/point/atom_types.py

**Comment by bergdorf** [comment]
Aha, very good, let me give this a try. Thanks for your help!

**Comment by bergdorf** [comment]
Small remark: if you change the number of hydrogens here: 
https://github.com/e3nn/e3nn/blob/master/examples/point/atom_types.py#L50
from 2 to 1 then you hit
 

```python
 File ".../lib/python3.7/site-packages/e3nn/rsh.py", line 120, in spherical_harmonics_alpha
    return out.reshape(*size, -1)  # [..., m]
RuntimeError: cannot reshape tensor of 0 elements into shape [0, -1] because the unspecified dimension size -1 can be any value and is ambiguous
```
due to the particulars of the filtering.

**Comment by mariogeiger** [comment]
fixed in 1365a7d8ab3cbaadddfbb8441c255912c231f24a
